File: 2.GammaCategorization/CreationalPatterns/5.Singleton/Aula07_SingletonExcercise/Aula_08_SinlgetonExercise2/POLICIA.java
import java.util.List;

public class POLICIA {
    public void enviarAlerta(String mensagem) {
        CentralDeAlertas.getInstancia().enviarAlerta("POLICIA", mensagem);
    }

    public List<String> listarAlertas() {
        return CentralDeAlertas.getInstancia().getAlertas();
    }
}

File: 2.GammaCategorization/CreationalPatterns/5.Singleton/Aula07_SingletonExcercise/Aula_08_SinlgetonExercise2/C#/POLICIA.cs
using System.Collections.Generic;

namespace CentralDeAlertasNamespace
{
    public class POLICIA
    {
        public void EnviarAlerta(string mensagem)
        {
            CentralDeAlertas.Instancia.EnviarAlerta("POLÍCIA", mensagem);
        }

        public IReadOnlyCollection<string> ListarAlertas()
        {
            return CentralDeAlertas.Instancia.GetAlertas();
        }
    }
}